tensorwaves/physics/helicityformalism/kinematics.py

```diff
@@ -1,16 +1,262 @@
+r"""Kinematic based calculations for the helicity formalism.
 
-class Kinematics():
-  def __init__(self):
-    pass
+It's responsibilities are defined by the interface
+:class:`.interfaces.Kinematics`.
 
-  def register_invariant_mass(self):
-    pass
+Here, the main responsibility is the conversion of general kinematic
+information of a reaction to helicity formalism specific quantities
 
-  def register_helicity_angles(self):
-    pass
+:math:`(s, \theta, \phi)`
 
-  def register_subsystem(self):
-    pass
+The basic building blocks are the :class:`~HelicityKinematics` and
+:class:`~SubSystem`.
+"""
 
-  def convert(events):
-    pass
\ No newline at end of file
+import logging
+from typing import Union
+
+import amplitf.kinematics as tfa_kin
+
+import numpy as np
+
+from tensorwaves.interfaces import Kinematics
+
+
+class SubSystem():
```

Comment:
How specific is the `SubSystem` to the helicity formalism? If not, it may be an idea to put it into the `physics` module itself.

Reply:
Good point. Should we move it to `physics/__init__.py`?

Reply:
Depends, see https://github.com/ComPWA/tensorwaves/pull/39/files#r415805711
In that case it could also be part of `formalism`. Not sure, depends where else a `SubSystem` is used.

Reply:
Let's leave it here for now an worry about it once an other formalism is implemented.